### PR TITLE
bug fix

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------
1. fix wrong operation stack bug [AB#1974846](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1974846)
2. update select status of subscriptions  [AB#1983740](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983740)
3. update message [AB#1948968](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?text=insigh&workitem=1948968)
4. [AB#1976033](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1976033)

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/OperationThreadContext.java [hunk 1]
```diff
@@ -55,6 +55,10 @@ public Operation currentOperation() {
         return this.operation;
     }
 
+    public void resetThreadId() {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/OperationThreadContext.java]: don't make it `public`

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java [hunk 2]
```diff
@@ -190,8 +190,8 @@ public void setSelectedSubscriptions(List<String> selectedSubscriptionIds) {
             throw new AzureToolkitRuntimeException("No subscriptions are selected. You must select at least one subscription.", IAccountActions.SELECT_SUBS);
         }
         final Set<String> selected = selectedSubscriptionIds.stream().map(String::toLowerCase).collect(Collectors.toSet());
-        this.getSubscriptions().forEach(s -> s.setSelected(false));
-        this.getSubscriptions().stream()
+        this.subscriptions.forEach(s -> s.setSelected(false));
+        this.subscriptions.stream()
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java]: does this change have any effects?

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/OperationThreadContext.java [hunk 3]
```diff
@@ -114,13 +114,19 @@ private synchronized void dispose() {
         final OperationThreadContext current = OperationThreadContext.current();
         final long threadId = Thread.currentThread().getId();
         assert this == current && this.threadId == threadId : String.format("[threadId:%s] disposing context[%s] in context[%s].", threadId, this, current);
-        if (this.parent == null || this.threadId != this.parent.threadId) { // this is the root task of current thread.
+        if (this.parent == null || this.threadId != this.parent.threadId || this.parent.threadId == -1) { // this is the root task of current thread.
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/OperationThreadContext.java]: should also check ancestors' threadId.

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**